src/components/TransactionImport/Form.tsx

```diff
@@ -29,23 +30,53 @@ const Form = ({ accounts, isLoading, setIsLoading, setResult }: Props) => {
         event.preventDefault()
 
         setIsLoading(true)
+        const file = (event.target as any).file.files[0] // TODO: without `any`
 
-        fetch(`/api/v1/import/ynab-import-preview?accountId=${accountId}`, {
-          method: 'POST',
-          body: new FormData(event.target as HTMLFormElement),
-        })
-          .then(checkStatus)
-          .then(parseJSON)
-          .then(body => {
-            setIsLoading(false)
-            if (error) {
-              setError('')
-            }
-            setResult(body.data, accountId)
+        parseFile(file)
+          .then(result => {
+            // Get the data and transform it to FormData for the backend
+            const data = new FormData()
+            data.append(
+              'file',
+              new File([new Blob([result[0].data])], 'ynap-transactions.csv', {
```

Comment:
To ensure we don't crash if `result` doesn't look as expected:

```suggestion
              new File([new Blob([result[0]?.data])], 'ynap-transactions.csv', {
```